scaloid-common/src/main/st/org/scaloid/common/app.scala

```diff
@@ -96,7 +108,17 @@ trait SActivity extends Activity with SContext with TraitActivity[SActivity] wit
 
   protected override def onCreate(b: Bundle) {
     super.onCreate(b)
-    onCreateBodies.foreach(_ ())
+    AppHelpers.createBundle.withValue(Option(b)) {
+      onCreateBodies.foreach(_ ())
+    }
+  }
+
+  def onCreate(body: Option[Bundle] => Any) = {
+    val el = { () =>
+      body(AppHelpers.createBundle.value)
```

Comment:
This is not changing the signature. If you look more closely at `Creatable`, it is built on a collection of *no-arg* functions because not all `onCreate()` Android methods take a `Bundle` as an argument. Further, the `Registerable` methods provide a pluggable way for external code to inject itself into `onCreate/onDestroy` or `onStart/onStop` based on needs, which also uses no-arg code blocks.

This change provides an alternative, not a changed method, which will only be used in the case where the user code actually needs a code thunk which does things with the `Bundle`, something specific to `Activity` subclasses. In this case, the value needs to be held out-of-band (in a `ThreadLocal`, or here, Scala's wrapper of it).